api/v1/verticadb_webhook.go

```diff
@@ -1765,6 +1769,85 @@ func (v *VerticaDB) checkNewSBoxOrSClusterShutdownUnset(allErrs field.ErrorList)
 	return allErrs
 }
 
+// checkShutdownForScaleUpOrDown ensures a subcluster to be scaled up/down has Shutdown field set to false and
+// live in a sandbox where Shutdown is set to false in spec/status for all pieces
+func (v *VerticaDB) checkShutdownForScaleUpOrDown(oldObj *VerticaDB, allErrs field.ErrorList) field.ErrorList {
+	newSclusterMap := v.GenSubclusterMap()
+	newSclusterSboxMap := v.GenSubclusterSandboxMap()
+	newSandboxMap := v.GenSandboxMap()
+	statusSclusterIndexMap := v.GenStatusSClusterIndexMap()
+	oldSclusterMap := oldObj.GenSubclusterMap()
+	newSclusterIndexMap := v.GenSubclusterIndexMap()
+	for subclusterName, sandboxName := range newSclusterSboxMap {
+		oldScluster, foundSclusterInOld := oldSclusterMap[subclusterName]
+		if !foundSclusterInOld {
+			continue // we only need to deal with persistent subclusters
+		}
+		newScluster, foundSclusterInNew := newSclusterMap[subclusterName]
+		if !foundSclusterInNew {
+			continue // avoid unexpected panic when something is wrong
+		}
+		sandbox := newSandboxMap[sandboxName]
+		if oldScluster.Size != newScluster.Size { // scale up/down
+			errMsgs := v.checkSboxForShutdown(sandbox, newSclusterMap, statusSclusterIndexMap)
+			if len(errMsgs) != 0 {
+				i := newSclusterIndexMap[subclusterName]
+				p := field.NewPath("spec").Child("subclusters").Index(i).Child("size")
+				err := field.Invalid(p,
+					newScluster.Size,
+					fmt.Sprintf("cannot scale up/down subcluster %q in sandbox %q because %q",
+						subclusterName, sandboxName, strings.Join(errMsgs, ",")))
+				allErrs = append(allErrs, err)
+			}
+		}
+	}
+	return allErrs
+}
+
+// checkSClusterToBeSandboxedShutdownUnset ensures the following when a subcluster (A) is to be added to a sandbox (B)
+// 1 A's Shutdown field is set to false
+// 2 B's Shutdown field is set to false
+// 3 B does not have any subcluster in it that has Shutdown set to true (spec/status)
+func (v *VerticaDB) checkSClusterToBeSandboxedShutdownUnset(allErrs field.ErrorList) field.ErrorList {
+	newSclusterMap := v.GenSubclusterMap()
+	statusScluterSboxMap := v.GenSubclusterSandboxStatusMap()
+	newSclusterSboxMap := v.GenSubclusterSandboxMap()
+	newSandboxIndexMap := v.GenSandboxIndexMap()
+	newSandboxMap := v.GenSandboxMap()
+	statusSclusterIndexMap := v.GenStatusSClusterIndexMap()
+	sandboxWithError := map[string]bool{}
+	for subclusterName, sandboxName := range newSclusterSboxMap {
```

Comment:
Minor: rename subclusterName to newSC, sandboxName to newSB for bettern readability.